llvm/cmake/modules/TableGen.cmake

```diff
@@ -19,19 +19,34 @@ function(tablegen project ofn)
 
   # Use depfile instead of globbing arbitrary *.td(s) for Ninja.
   if(CMAKE_GENERATOR MATCHES "Ninja")
```

Comment:
```
if(CMAKE_GENERATOR MATCHES "Ninja" AND cmp0116_state STREQUAL "NEW")
```

I think it would be fair to fall back to `file(GLOB)` if `CMP0116` is `OLD`.